lib/fourOhFour.js

```diff
@@ -126,25 +127,12 @@ function fourOhFour (options) {
 
     avvio.once('preReady', () => {
       const context = this[kFourOhFourContext]
-
-      const onRequest = this[kHooks].onRequest
-      const preParsing = this[kHooks].preParsing.concat(opts.preParsing || [])
-      const preValidation = this[kHooks].preValidation.concat(opts.preValidation || [])
-      const preSerialization = this[kHooks].preSerialization.concat(opts.preSerialization || [])
-      const preHandler = this[kHooks].preHandler.concat(opts.preHandler || [])
-      const onSend = this[kHooks].onSend
-      const onError = this[kHooks].onError
-      const onResponse = this[kHooks].onResponse
-
-      context.onRequest = onRequest.length ? onRequest : null
-      context.preParsing = preParsing.length ? preParsing : null
-      context.preValidation = preValidation.length ? preValidation : null
-      context.preSerialization = preSerialization.length ? preSerialization : null
-      context.preHandler = preHandler.length ? preHandler : null
-      context.onSend = onSend.length ? onSend : null
-      context.onError = onError.length ? onError : null
-      context.onResponse = onResponse.length ? onResponse : null
-
+      for (const hook of lifecycleHooks) {
+        const toSet = this[kHooks][hook]
+          .concat(opts[hook] || [])
+          .map(h => h.bind(this))
+        context[hook] = toSet.length ? toSet : null
+      }
```

Comment:
Can you move this logic into the `hooks.js` file?  It seems way more useful to share this, also because it's duplicated in `route.js`.

Reply:
Not sure what you mean, for having this working we need to work with this closure.
Moving to an external file will only make the core a little more complicated for saving 6 lines. Furthermore, now with the for loop and the `lifecycleHooks` array the code is more future proof, as if we will add new hooks, this code will continue to work as expected and handle the new hook as well.

Reply:
let’s land this, will fix if it becomes an issue